modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java

```diff
@@ -776,6 +777,10 @@ private void _addOrUpdateNestedObjectEntries(
 						objectDefinition, objectRelationship, primaryKey,
 						nestedObjectEntry.getId());
 				}
+
+				if (!nestedObjectEntries.isEmpty()) {
+					NestedFieldsSupplier.addFieldName(entry.getKey());
+				}
```

Comment:
Maybe it is a good moment to update the `if (relatedObjectDefinition.isSystem()) {` validation for `if (relatedObjectDefinition.isUnmodifiableSystemObject()) {`

Reply:
Yeah, we disccused it now

Reply:
I will create a separate ticket for that, and review all the usages in one go.

Reply:
https://issues.liferay.com/browse/LPS-184044